### PR TITLE
add DPoP encoding utils

### PR DESCRIPTION
**Pull request by jshawl**
This PR implements a few utility functions that will be used in an upcoming PR for jwt creation and signatures as part of the larger [DPoP](https://datatracker.ietf.org/doc/html/rfc9449) initiative that will be used in the hosted buttons component. 

I am planning on two follow up prs to this repo:

- one that implements the key pair generation and signing and verification
- one that implements the jwt creation and exports a `dPoPHeaders` function that will be used in checkout-components.

### CODE REVIEW

File: src/dpop.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    .replace(/\//g, "_");
+};
+
+export const base64decodeUrlSafe = (string: string): string => {
+  return atob(string.replace(/-/g, "+").replace(/_/g, "/"));
+};
+
+export const sha256 = (string: string): ZalgoPromise<string> => {
+  const bytes = stringToBytes(string);
+  return window.crypto.subtle.digest("sha-256", bytes).then((digest) => {
```
Comment (wsbrunson) [inline src/dpop.js:28]: You could make this a regular promise, since this `.then` would negate any of the synchronous abilities of Zalgo.

Comment (wsbrunson) [reply src/dpop.js:28]: and if so, i'd suggest going with `async/await`

**Review by wsbrunson — APPROVED** [review]

File: src/dpop.js (hunk 1 above)
Comment (jshawl) [reply src/dpop.js:28]: amazing! would you `eslint-disable` these?

```
24:47  error  Unexpected use of 'Promise'  no-restricted-globals
24:47  error  "Promise" is not defined     promise/no-native
```

it looks like it is explicitly enabled here - https://github.com/krakenjs/grumbler-scripts/blame/3d70c3c2a809793b420fadf4c6859016fdcbb094/config/.eslintrc-browser.js#L8-L9 🤔

Comment (wsbrunson) [reply src/dpop.js:28]: yeah i usually disable them at the top of the file

**Pull request merged**